snuba/datasets/sessions.py

```diff
@@ -0,0 +1,259 @@
+    def process_message(self, message, metadata=None) -> Optional[ProcessedMessage]:
+        if message["duration"] is None:
+            duration = None
+        else:
+            duration = _collapse_uint32(int(message["duration"] * 1000))
+
+        # since duration is not nullable, the max duration means no duration
+        if duration is None:
+            duration = MAX_UINT32
+
+        processed = {
+            "session_id": str(uuid.UUID(message["session_id"])),
+            "distinct_id": str(uuid.UUID(message.get("distinct_id") or NIL_UUID)),
+            "seq": message["seq"],
+            "org_id": message["org_id"],
+            "project_id": message["project_id"],
+            "retention_days": message["retention_days"],
+            "duration": duration,
+            "status": STATUS_MAPPING[message["status"]],
+            "errors": _collapse_uint16(message["errors"]) or 0,
+            "received": _ensure_valid_date(
+                datetime.utcfromtimestamp(message["received"])
+            ),
+            "started": _ensure_valid_date(
+                datetime.utcfromtimestamp(message["started"])
+            ),
+            "release": message["release"],
+            "environment": message.get("environment") or "",
+        }
+        return ProcessedMessage(action=ProcessorAction.INSERT, data=[processed])
+
+
+class SessionsDataset(TimeSeriesDataset):
+    def __init__(self) -> None:
+        all_columns = ColumnSet(
+            [
+                ("session_id", UUID()),
+                ("distinct_id", UUID()),
+                ("seq", UInt(64)),
+                ("org_id", UInt(64)),
+                ("project_id", UInt(64)),
+                ("retention_days", UInt(16)),
+                ("duration", UInt(32)),
+                ("status", UInt(8)),
+                ("errors", UInt(16)),
+                ("received", DateTime()),
+                ("started", DateTime()),
+                ("release", LowCardinality(String())),
+                ("environment", LowCardinality(String())),
+            ]
+        )
+
+        write_schema = MergeTreeSchema(
+            columns=all_columns,
+            local_table_name=WRITE_LOCAL_TABLE_NAME,
+            dist_table_name=WRITE_DIST_TABLE_NAME,
+            order_by="(org_id, project_id, release, environment, started)",
+            partition_by="(toMonday(started))",
+            settings={"index_granularity": 16384},
+        )
+
+        read_columns = ColumnSet(
+            [
+                ("org_id", UInt(64)),
+                ("project_id", UInt(64)),
+                ("started", DateTime()),
+                ("release", LowCardinality(String())),
+                ("environment", LowCardinality(String())),
+                (
+                    "duration_quantiles",
+                    AggregateFunction("quantilesIf(0.5, 0.9)", UInt(32), UInt(8)),
+                ),
+                ("uniq_sessions", AggregateFunction("countIf", UUID(), UInt(8))),
```

Comment:
Any reason for not calling this count_sessions or sessions? uniq_sessions may make people believe this is a uniq aggregation.